### PR TITLE
Add bash and zsh completion to homebrew formula

### PR DESCRIPTION
**Pull request by eddiezane**
### Description

This PR adds a [custom install block](https://goreleaser.com/customization/homebrew/) to the goreleaser brew config to install bash and zsh completion.

I tested these changes manually in a direct edit of the brew formula. I think goreleaser updated their config format here to be arrays (brews: - / repos: -) instead of the current singular (brew: / repo:) so this may need a larger patch to make the config compliant.

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [x] Make sure the title of the PR is a good description that can go into the release notes

/area build-and-release

### CODE REVIEW

File: .goreleaser.brew.yml [hunk 1]
```diff
@@ -20,4 +20,15 @@ brew:
...
+
+  install: |
+      bin.install "eksctl"
+
+      # Install bash completion
+      output = Utils.popen_read("#{bin}/eksctl completion bash")
+      (bash_completion/"eksctl").write output
+
+      # Install zsh completion
+      output = Utils.popen_read("#{bin}/eksctl completion zsh")
```
Comment (sayboras) [inline .goreleaser.brew.yml:33]: just curious how it's working in general and in the case that user didn't have zsh ?

Comment (eddiezane) [reply .goreleaser.brew.yml:33]: Homebrew convention is to always install bash/zsh completion when available regardless of whether the user has the shell installed or not. Default location for zsh is `/usr/local/share/zsh/site-functions`.

This is copied from the formula for kubectl https://github.com/Homebrew/homebrew-core/blob/a8e1b997192a49fc974eeeb914028abea1c6bb80/Formula/kubernetes-cli.rb#L32-L38.

Comment (sayboras) [reply .goreleaser.brew.yml:33]: thanks for your clarification, it's good to know :100:

File: .goreleaser.brew.yml [hunk 2]
```diff
@@ -20,4 +20,15 @@ brew:
     - aws-iam-authenticator
 
   custom_block: |
-    head "https://github.com/weaveworks/eksctl/releases/download/latest_release/eksctl_Darwin_amd64.tar.gz"
+    head "https://github.com/weaveworks/eksctl/releases/download/latest_release/eksctl_Darwin_amd65.tar.gz"
```
Comment (michaelbeaumont) [inline .goreleaser.brew.yml:23]: `65`

Comment (eddiezane) [reply .goreleaser.brew.yml:23]: 👍 rebased

Comment (michaelbeaumont) [reply .goreleaser.brew.yml:23]: It's still changed to `amd65`, that can't be right. accidental vim ctrl-a?

Comment (eddiezane) [reply .goreleaser.brew.yml]: Ahhh sorry. Totally misunderstood your original comment. 

Good catch - must have been ctrl-a (which is also my tmux bind).

Fixed.

**Pull request merged**